hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala

```diff
@@ -796,23 +797,52 @@ object LowerTableIR {
           context,
           ctxRef => ToStream(ctxRef, true))
 
-      case t@TableGen(_, _, cname, gname, _, partitioner, traceId) =>
-        val Seq(contexts, globals, body) = t.children.map(ir => lowerIR(ir.asInstanceOf[IR]))
+      case TableGen(contexts, globals, cname, gname, body, partitioner, errorId) =>
+        val loweredGlobals = lowerIR(globals)
         TableStage(
-          globals,
+          loweredGlobals,
           partitioner = partitioner,
           dependency = TableStageDependency.none,
-          // Assert at runtime that the number of contexts matches the number of partitions
-          contexts = bindIR(ToArray(contexts)) { ref =>
-            val dieMsg = strConcat(
-              s"${t.getClass.getSimpleName}: partitioner contains ${partitioner.numPartitions} partitions, got ",
-              invoke("str", TString, ArrayLen(ref))
-            )
-            ToStream(
-              If(ArrayLen(ref) ceq partitioner.numPartitions, ref, Die(dieMsg, ref.typ, traceId))
-            )
+          contexts = lowerIR {
+            bindIR(ToArray(contexts)) { ref =>
+              bindIR(ArrayLen(ref)) { len =>
+                // If contexts were an array we could assert this at compile time...
```

Comment:
not true -- arrays aren't statically sized in the general case. We could write an analysis that tries to infer static length, but not worth it to eliminate this check.

Reply:
But would those arrays not be fixed size generated from python?

Reply:
you're right that if someone is passing in a partitioner, someone at some point probably is generating an array of a fixed size. But we don't know the runtime length of any particular IR of type TArray by default.

Reply:
fair enough!